src/interactivity/win32/uiaTextRange.cpp

```diff
@@ -131,25 +129,26 @@ IFACEMETHODIMP UiaTextRange::FindText(_In_ BSTR text,
 
         CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
         THROW_HR_IF(E_POINTER, !gci.HasActiveOutputBuffer());
-        Search searcher{ gci.renderData, wstr, searchDirection, sensitivity, _endpointToCoord(_pData, searchAnchor) };
+        Search searcher{ gci.renderData, wstr, searchDirection, sensitivity, searchAnchor };
 
         HRESULT hr = S_OK;
         if (searcher.FindNext())
         {
             const auto foundLocation = searcher.GetFoundLocation();
-            const Endpoint start = _coordToEndpoint(_pData, foundLocation.first);
-            const Endpoint end = _coordToEndpoint(_pData, foundLocation.second);
+            const auto start = foundLocation.first;
+            const auto end = foundLocation.second;
+            const auto bufferSize = _pData->GetTextBuffer().GetSize();
+
             // make sure what was found is within the bounds of the current range
-            if ((searchDirection == Search::Direction::Forward && end < _end) ||
-                (searchDirection == Search::Direction::Backward && start > _start))
+            if ((searchDirection == Search::Direction::Forward && bufferSize.CompareInBounds(end, _end) < 0) ||
+                (searchDirection == Search::Direction::Backward && bufferSize.CompareInBounds(start, _start) > 0))
             {
                 hr = Clone(ppRetVal);
                 if (SUCCEEDED(hr))
                 {
                     UiaTextRange& range = static_cast<UiaTextRange&>(**ppRetVal);
                     range._start = start;
                     range._end = end;

```

Comment:
make sure you're maintaining exclusivity.

Reply:
This is actually going to be fixed in https://github.com/microsoft/terminal/pull/4373 (which targets accessibility _after_ the refactor).